### PR TITLE
[curl] Fix SSL on iOS

### PR DESCRIPTION
**Pull request by redbaron**
Enables secure transport SSL backend on iOS.

When CURL_CA_BUNDLE is set (defaults to /etc/ssl/cert.pem) curl sets it as CA for every request, but on iOS this file is missing and curl's `sectransp` SSL implementation can't deal with it, failing the request completely.

This change makes CURL_CA_BUNDLE to be unset, much like CURL_CA_PATH so that `sectransp` uses system CA store.

**Describe the pull request**

- #### What does your PR fix?

Fixes curl  SSL support on iOS

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
all , no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
yes

### CODE REVIEW

**Review by FrankXie05 — DISMISSED** [review]

File: ports/curl/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,7 @@
 {
   "name": "curl",
   "version": "7.84.0",
+  "port-version": 2,
```
Comment (Thomas1664) [inline ports/curl/vcpkg.json:4]:
```suggestion
  "port-version": 1,
```

File: versions/baseline.json [hunk 2]
```diff
@@ -1770,7 +1770,7 @@
     },
     "curl": {
       "baseline": "7.84.0",
-      "port-version": 0
+      "port-version": 2
```
Comment (FrankXie05) [inline versions/baseline.json:1773]:
```suggestion
      "port-version": 1
```
Please bump the port version. See [documentation](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#update-the-version-field-in-the-control-file-of-any-modified-ports).

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

File: versions/c-/curl.json [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    {
+      "git-tree": "4d64a4ac7f080159be045ccbf0a7fa81ef859cfa",
+      "version": "7.84.0",
+      "port-version": 1
+    },
+    {
+      "git-tree": "90b69f8fa78155471d8079cdfd7a8b889c9971df",
+      "version": "7.84.0",
+      "port-version": 2
+    },
```
Comment (FrankXie05) [inline versions/c-/curl.json:12]:
```suggestion
```

**Review by FrankXie05 — CHANGES_REQUESTED** [review]

**Review by FrankXie05 — APPROVED** [review]

**Comment by ras0219-msft** [comment]
This makes sense to me. Thanks for the improvement!

**Pull request merged**